Enable connecting to metrics without certificates

npwalker opened the pull request:
Prior to this commit, we used certificates to connect to the metrics
endpoints despite using SSL_VERIFY_NONE.

After this commit, we forgo the certificates so non-root users can easily
run the script.

reidmv commented:
@npwalker have you had a chance to test this yet?

npwalker commented:
@reidmv I came to a different conclusion after testing.  No need to use certs at all.

npwalker committed:
Add 4.3.0 to changelog and bump version in metadata.json

jarretlavallee reviewed (approved):
:+1: Looks good on Davis and Glisan

npwalker commented:
Tested this out a few different times.  

Essentially running the script as the centos user ensure's that it can be run as non-root.  Due to the 744 permissions of the script we execute it via ruby instead of executing the script directly.  

```
su - centos -c "/opt/puppetlabs/puppet/bin/ruby /opt/puppetlabs/pe_metric_curl_cron_jobs/scripts/tk_metrics --metrics_type puppetserver"

su - centos -c "/opt/puppetlabs/puppet/bin/ruby /opt/puppetlabs/pe_metric_curl_cron_jobs/scripts/tk_metrics --metrics_type puppetdb"
```

npwalker committed:
No longer pass certificate information to netHTTP

Prior to this commit, we set the cacert, cert, and key but also
pass SSL_VERIFY_NONE which negates the need for the former.

After this commit, we do not set the cacert, cert, and key so that
non-root users can run the script without hitting permissions
errors on the keys.
